receiver/kubeletstatsreceiver/runnable.go

```diff
@@ -89,7 +91,7 @@ func (r *runnable) Run() error {
 	ctx := obsreport.ReceiverContext(r.ctx, typeStr, transport, r.receiverName)
 	for _, md := range mds {
 		ctx = obsreport.StartMetricsReceiveOp(ctx, typeStr, transport)
-		err = r.consumer.ConsumeMetricsData(ctx, *md)
+		err = r.consumer.ConsumeMetrics(ctx, pdatautil.MetricsFromMetricsData([]consumerdata.MetricsData{*md}))
```

Comment:
Please create an issue to change line 89 kubelet.NewMetadata(r.extraMetadataLabels, podsMetadata) to return an array of non pointers to avoid this for loop :)

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/703